smartsim/_core/control/controller.py

```diff
@@ -527,6 +555,7 @@ def _launch_step(
             entity.name not in self._jobs.jobs and entity.name not in self._jobs.db_jobs
         ):
             try:
+                self.symlink(job_step, entity)
```

Comment:
The call to `symlink` is in a `try`-`except` clause, but we don't actually catch any filesystem related exception. Should we catch such exceptions (possibly raising them with some informative output), or should we remove the call from the block?

Reply:
I think there are definitely some exceptions that could be raised so I'll add them in with some informative output. Thanks for catching that!

Reply:
I removed the call from the call block and handle the exception within `symlink_output_files`. Let me know what you think about it!